common/network-common/src/main/java/org/apache/spark/network/TransportContext.java

```diff
@@ -144,14 +161,17 @@ public TransportChannelHandler initializePipeline(
       RpcHandler channelRpcHandler) {
     try {
       TransportChannelHandler channelHandler = createChannelHandler(channel, channelRpcHandler);
+      ChunkFetchRequestHandler chunkFetchHandler = createChunkFetchHandler(channelHandler, channelRpcHandler);
       channel.pipeline()
         .addLast("encoder", ENCODER)
         .addLast(TransportFrameDecoder.HANDLER_NAME, NettyUtils.createFrameDecoder())
         .addLast("decoder", DECODER)
         .addLast("idleStateHandler", new IdleStateHandler(0, 0, conf.connectionTimeoutMs() / 1000))
         // NOTE: Chunks are currently guaranteed to be returned in the order of request, but this
         // would require more logic to guarantee if this were not part of the same event loop.
-        .addLast("handler", channelHandler);
+        .addLast("handler", channelHandler)
+        // Use a separate EventLoopGroup to handle ChunkFetchRequest messages.
+        .addLast(chunkFetchWorkers, "chunkFetchHandler", chunkFetchHandler);
```

Comment:
Hmm... I think there is some waste here. Not all channels actually need the chunk fetch handler. Basically only the shuffle server (external or not) does. So for all other cases - RpcEnv server and clients, shuffle clients - you'd have this new thread pool just sitting there. Before you'd have the handler in the pipeline, but that's just a few bytes of memory being wasted.

It would be good to avoid that.

Reply:
yes i did notice that... makes sense